### ISSUE TITLE
Unclear in API when ID is a number in "switch to frame"

### ISSUE THREAD

**Issue by mercmobily** [w3c/webdriver#1205]
The section [switch to frame](https://w3c.github.io/webdriver/webdriver-spec.html#switch-to-frame) in the specs is a little unclear in regards of what "ID" is supposed to be.
According to the specs:

> If id is not null, a Number object, or an Object that represents a web element, return error with error code no such frame.

While `id is null` and `id represents a web element` are clear, `id is a Number object` leaves me with the question: "which number"? The link to [supported properties index](https://w3c.github.io/webdriver/webdriver-spec.html#dfn-supported-property-index) points to the [IDL specs' cover of supported property index](https://heycam.github.io/webidl/#dfn-supported-property-indices). 

**So... where do I get this number from, exactly, if I wanted to use to switch frame?**

Please remember that I am not talking about programming languages API -- I am talking about the usage of the API without support library. I am specifying this because the IDL link seems to be all about Java, whereas my question is on the use of the Webdriver endpoints.

**Comment by andreastt** [comment]
The input can be either `{"id": null}` which puts you back in the top-level browsing context, `{"id": 2}` which gets the Nth frame from the `window.frames` array, or `{"id": {"element-6066-11e4-a52e-4f735466cecf": "<UUID>"}}` which puts you in the context of that frame element.

I think talking about “Number” as an “object” is misleading, as it is a primitive in JS.

I also think the spec _means_ to talk about the support property index of `window.frame`, not the `window` global itself. Maybe @AutomatedTester can confirm?

**Comment by mercmobily** [comment]
Hi,

I guess the key sentence I didn't understand was "Let window be the associated window of the current browsing context’s active document. If id is not a supported property index of window, return error with error code no such frame." . Here, _associated window of the current browsing context’s active document_  basically means " the Nth frame from the window.frames array" ...right?

If so, it would be _grand_ if something practical like this was specified in the specs... Because that sentence is really obscure. Closing not to pollute -- but please don't forget.

**Comment by andreastt** [comment]
Yes, that sentence does not make any sense to me. I don’t know what an _associated window_ is either.

Instead of using the `WindowProxy`’s [[[GetOwnProperty]]](https://html.spec.whatwg.org/#windowproxy-getownproperty) this should be using the underlying definitions in HTML for indexing the document-tree child browsing contexts of the document’s browsing context.

Also I was mistaken in my last reply about `window.frames`, as this just returns the `WindowProxy`. It is the `WindowProxy` itself that one should be indexing.

It is clear there is a grave need for cleanup of this command.

**Comment by mercmobily** [comment]
Sorry for closing -- I was honestly, frankly thinking that that sentence
didn't make sense because I was too stupid to understand that. It wouldn't
be the first time.
I wish I could fix it myself... Sorry but  I can't help very much. I am
just a humble developer trying to write a [clear, down-to-earth,
async-await-based, fully compliant API](
https://github.com/mercmobily/webrider) which will hopefully become the
port of call for people who want to use the protocol easily. 3 classes and
500 SLOC rather than 93 classes and ~5000 SLOC from the official Selenium
API...
